Sources/PrimerSDK/Classes/User Interface/TokenizationViewModels/FormPaymentMethodTokenizationViewModel.swift

```diff
@@ -965,32 +918,80 @@ extension FormPaymentMethodTokenizationViewModel {
                 }
                 .catch { err in
                     DispatchQueue.main.async {
-                        Primer.shared.delegate?.onResumeError?(err)
+                        self.handleErrorBasedOnSDKSettings(err, isOnResumeFlow: true)
                     }
-                    self.handle(error: err)
                 }
             } else {
                 let err = PrimerError.invalidValue(key: "resumeToken", value: nil, userInfo: ["file": #file, "class": "\(Self.self)", "function": #function, "line": "\(#line)"])
                 ErrorHandler.handle(error: err)
-                
-                handle(error: err)
                 DispatchQueue.main.async {
-                    PrimerDelegateProxy.onResumeError(err)
+                    self.handleErrorBasedOnSDKSettings(err, isOnResumeFlow: true)
                 }
             }
         }
     }
 }
 
+extension FormPaymentMethodTokenizationViewModel {
+    
+    private func dispatchActions() -> Promise<Void> {
+        
+        return Promise { seal in
+            
+            var network = self.cardNetwork?.rawValue.uppercased()
+            if network == nil || network == "UNKNOWN" {
+                network = "OTHER"
+            }
+            
+            let params: [String: Any] = [
+                "paymentMethodType": "PAYMENT_CARD",
+                "binData": [
+                    "network": network,
+                ]
+            ]
+                
+            var actions = [ClientSession.Action(type: "SELECT_PAYMENT_METHOD", params: params)]
+            
+            if (requirePostalCode) {
+                let state: AppStateProtocol = DependencyContainer.resolve()
+                
+                let currentBillingAddress = state.primerConfiguration?.clientSession?.customer?.billingAddress
+                
+                let billingAddressParams = [
+                    "firstName": currentBillingAddress?.firstName as Any,
+                    "lastName": currentBillingAddress?.lastName as Any,
+                    "addressLine1": currentBillingAddress?.addressLine1 as Any,
+                    "addressLine2": currentBillingAddress?.addressLine2 as Any,
+                    "city": currentBillingAddress?.city as Any,
+                    "postalCode": postalCodeField.postalCode,
+                    "state": currentBillingAddress?.state as Any,
+                    "countryCode": currentBillingAddress?.countryCode as Any
```

Comment:
Same as my other comment, I believe this line would crash, did you test it?

`"countryCode": currentBillingAddress?.countryCode as Any`